meilisearch-release-changelog/v1.2.0.md

```diff
@@ -0,0 +1,203 @@
+
+v1.2 brings a massive refactor of the search engine that brings perforance and relevancy improvements. Also, one of the major benefits of this refactoring is the ability to explore and integrate new features related to the search in the future releases. Stay tuned!
+
+Most of the points linked below can be really technical. If you have any questions or unexpected behaviors on your side, feel free to [reach us](https://github.com/meilisearch/meilisearch/discussions).
+
+Done by @loiclec, @dureuill, @ManyTheFish, @Kerollmops and @irevoire in #3542.
+
+### Performance improvements
+
+- The fastest 75 percent of queries now consistently answer below 50ms on our test setup.
+- The complexity of a search query is now limited, in order to limit CPU time and RAM consumption. In more details:
+  - The number of typo considered for a single term is now limited to 150 for 1 typo, 50 for 2 typos. Similarly, the number of considered (single word or multi word) synonyms is limited to 50, the total number of words for all synonyms of a term is limited to 100.
+  - The number of words considered in a query is now limited to 1000. The number of terms (words not inside a phrase, and individual phrases) was and is still limited to 10, so this change only impacts queries containing very long phrases.
+- Geo search improvements
+  - Improved performance when there are only a few documents to geosort.
+  - Descending sort is now as performant as ascending sort.
+- Solves some performance issues: https://github.com/meilisearch/meilisearch/issues/3124, https://github.com/meilisearch/meilisearch/issues/3378, https://github.com/meilisearch/meilisearch/issues/2202, https://github.com/meilisearch/meilisearch/issues/3123
+
+### Relevancy improvements
+
+- Synonyms are no longer considered as exact matches by the `exactness` ranking rule. This change is due to the goal of the `exactness` ranking rule of boosting the documents that contain the query exactly as typed by the user. Other ranking rules will still consider synonyms. Solves https://github.com/meilisearch/meilisearch/issues/3410
+- `Attributes`, `Exactness`, `typo` and `proximity` ranking rules now imply `words` if `words` was not already specified before them. This change was made because the behavior of applying these ranking rules before or without `words` was unintuitive and could hurt relevancy.
+- Split words are now looked for for digrams. This means that the query `whit ehorse` may now match documents containing `white horse`, which improves the relevancy of typo tolerance.
+- ngrams and split words are now disadvantaged in comparison to the exact word in the `typo` ranking rule.
+- The "iterative" version of ranking rules are now removed (https://github.com/meilisearch/meilisearch/issues/3356). The iterative version of algorithms would be applied depending on internal factors and would sometimes differ greatly in relevancy from the base ranking rule, leading to inconsistent relevancy.
```

Comment:
Pretty sure the users don't know what is the "iterative" version of the ranking rules. Should we keep it?

Reply:
We may keep it, but should definitely shorten it. I'm making a ocnservative suggestion here, but we could even remove everything after the issue link.

```suggestion
- Removed iterative version of ranking rules (https://github.com/meilisearch/meilisearch/issues/3356). Meilisearch used the iterative version of algorithms depending on internal factors and could lead to inconsistent relevancy ranking
```

Reply:
No understandable impact from the user's point of view. The behaviour is now just more consistent in terms of both relevancy and latency.

Reply:
Shorter suggestion adapting @loiclec's comment and focusing on user impact:

```suggestion
- Removed iterative version of ranking rules, improving consistency in terms of both relevancy and latency (https://github.com/meilisearch/meilisearch/issues/3356) 
```

Reply:
I think @dureuill's current suggestion is great, actually. Unless we can explain iterative functions in one sentence, I'd say it's out of scope for a changelog.